### PR TITLE
init project

### PR DESCRIPTION
**Pull request by aminerhanemi**
by running npm build the ts files in  the backend and the public will be compiled and added to the hello-world.lke folder .
hello-world.lke  folder is the one we need to compress and add to the server.

Missing thing:

- readme content 
- a script to compress the main folder (Already done by andrea )
- a way to avoid copying the file each time to the server to test
- ....

### CODE REVIEW

File: hello-world.lke/buildcache/backend.tsbuildinfo [hunk 1]
```diff
@@ -0,0 +1,1730 @@
+{
```
Comment (yigitakbulut) [inline hello-world.lke/buildcache/backend.tsbuildinfo]: What is the purpose of this file? I think we can add this into .gitignore  as well.

File: hello-world.lke/buildcache/public.tsbuildinfo [hunk 2]
```diff
@@ -0,0 +1,210 @@
+{
```
Comment (yigitakbulut) [inline hello-world.lke/buildcache/public.tsbuildinfo]: This is weird, it is in the .gitignore but still pushed.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,5 @@
+
+.npmrc
+node_modules
+.idea
+hello-world.lke/buildcache/public.tsbuildinfo
```
Comment (francescoinfante) [inline .gitignore]: just use *.lke, so the .gitignore will work also as boilerplate

File: hello-world.lke/backend/routeHandler.d.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import { PluginConfig, PluginRouteOptions } from '../@types/plugin';
```
Comment (francescoinfante) [inline hello-world.lke/backend/routeHandler.d.ts]: we should not push the *.lke file to git. this is built code

File: src/public/index.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+function main() {
```
Comment (francescoinfante) [inline src/public/index.ts:1]: when does this code get executed?

Instead I would just register a backend API that return hello world

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+    "include": [], // DO NOT DELETE: mandatory to keep even if empty
```
Comment (francescoinfante) [inline tsconfig.json:2]: why?

**Review by francescoinfante — CHANGES_REQUESTED** [review]

File: package.json [hunk 7]
```diff
@@ -1,36 +1,41 @@
...
+    "copy-assets": "cp -R src/public/assets hello-world.lke/public && cp src/public/index.html hello-world.lke/public",
+    "build:backend": "tsc -b src/backend && npm run installProduction",
+    "installProduction": "cp package.json hello-world.lke && cd hello-world.lke && npm install --production",
+    "copyManifest": "cp -R src/manifest.json hello-world.lke/",
     "test": "echo \"Error: no test specified\" && exit 1"
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/Linkurious/lke-plugin-hello-world.git"
+    "url": "git+https://github.com/Linkurious/lke-plugin-image-export.git"
```
Comment (aminerhanemi) [inline package.json]: hello-world and not image-export (same in all the file )

**Pull request merged**